src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -83,6 +83,12 @@ public function createTemplate(UI\Control $control = NULL)
 		$latte->addFilter('modifyDate', function ($time, $delta, $unit = NULL) {
 			return $time == NULL ? NULL : Nette\Utils\DateTime::from($time)->modify($delta . $unit); // intentionally ==
 		});
+		$latte->addFilter('translate', function ($text/*, ...$params*/) use ($template) {
+			if (($translator = $template->getTranslator()) && $translator instanceof Nette\Localization\ITranslator) {
+				return call_user_func_array([$translator, 'translate'], func_get_args());
+			}
+			throw new Nette\InvalidStateException('Translator has not been set.');
+		});
```

Comment:
**ad. 1st && 2nd point**: Fixed
**ad. 3rd point**: I agree with that, but when should be this check executed? You can never know when `ITranslator` will be passed to `Template`.

My only idea about this is something simlar like:

```php
$latte->addFilter('translate', function ($text, ...$params) use ($latte, $template) {
	if ($translator = $template->getTranslator()) {
		$latte->addFilter('translate', function ($text, ...$params) use ($template) {
			return call_user_func_array([$translator, 'translate'], func_get_args());
		}
		return $template->translate($text, ...$params));
	}
	throw new Nette\InvalidStateException('Translator has not been set.');
});

```

This breaks in situations, where you set different `Translator` in the middle of running app (but I think thats ok, I cannot think about any use-case)